ddtrace/tracer/rules_sampler.go

```diff
@@ -54,6 +54,55 @@ func (r *rulesSampler) HasSpanRules() bool { return r.spans.enabled() }
 
 func (r *rulesSampler) TraceRateLimit() (float64, bool) { return r.traces.limit() }
 
+type Provenance int32
+
+const (
+	Local    Provenance = iota
+	Customer Provenance = 1
+	Dynamic  Provenance = 2
+)
+
+var (
+	ProvenanceName = map[Provenance]string{
+		Local:    "local",
+		Customer: "customer",
+		Dynamic:  "dynamic",
+	}
+	ProvenanceValue = map[string]Provenance{
+		"local":    Local,
+		"customer": Customer,
+		"dynamic":  Dynamic,
+	}
+)
+
+func (p Provenance) String() string {
+	return ProvenanceName[p]
+}
+
+func (p Provenance) MarshalJSON() ([]byte, error) {
+	return json.Marshal(p.String())
+}
+
+func (p *Provenance) UnmarshalJSON(data []byte) error {
+	var prov string
+	var err error
+	if err = json.Unmarshal(data, &prov); err != nil {
+		return err
+	}
+	if *p, err = ParseProvenance(prov); err != nil {
+		return err
+	}
+	return nil
+}
+
+func ParseProvenance(p string) (Provenance, error) {
+	v, ok := ProvenanceValue[strings.TrimSpace(strings.ToLower(p))]
+	if !ok {
+		return Customer, fmt.Errorf("Invalid Provenance: \"%v\"", p)
+	}
+	return v, nil
+}
```

Comment:
```suggestion
func ParseProvenance(p string) (Provenance, error) {
	for _, v := range Provenances {
		if strings.EqualFold(strings.TrimSpace(strings.ToLower(p)), p) {
			return v, nil
		}
	}
	return Customer, fmt.Errorf("Invalid Provenance: \"%v\"", p)

}
```

Reply:
Thanks for the suggestions. Can you tell me what the difference is?

Reply:
We don't need unnecessary error prone maps, code will be cleaner and easier / better to maintain

Reply:
this is where I'm not getting. can you let me know why the maps are error prone? They map into each other.